zookeeper-server/src/main/java/org/apache/zookeeper/common/X509Util.java

```diff
@@ -81,7 +87,29 @@ public abstract class X509Util implements Closeable, AutoCloseable {
         }
     }
 
-    public static final String DEFAULT_PROTOCOL = "TLSv1.2";
+    public static final String DEFAULT_PROTOCOL = defaultTlsProtocol();
+
+    /**
+     * Return TLSv1.3 or TLSv1.2 depending on Java runtime version being used.
+     * TLSv1.3 was first introduced in JDK11 and back-ported to OpenJDK 8u272.
+     */
+    private static String defaultTlsProtocol() {
+        try {
+            List<String> supported = Arrays.asList(SSLContext.getDefault().getSupportedSSLParameters().getProtocols());
+            if (supported.contains("TLSv1.3")) {
+                return "TLSv1.3";
```

Comment:
What about logging a line here ?

"Default TLS protocol is {}, supported TLS protocols are {}"

Reply:
I've added log printout.